### DISCUSSION TITLE
Smart Corners - Interpolation issue

### DISCUSSION THREAD

**Discussion by Wirdek** [kateliev/TypeRig#58] [Ideas]
Hello Vassil,
I want to discuss one Fontlab issue - I have already addressed this to Fontlab about year ago but there is no progress since then.
_(... This feature request is registered in our database on GitHub as #5441. It is in the list of other 800+ issues including both bugs and features. We have to prioritise. Unfortunately I cannot say when and if it will be implemented.)_

I would like to make "smart corners" a bit smarter.
No big deal – just to prevent merging/welding of affected adjacent points.
It would geatly improve interpolation possibilities. I guess the image is self-explanatory:

![image](https://user-images.githubusercontent.com/68431285/134067526-8e3d9698-b79b-4d13-ab57-307aa46575e5.png)

What do you think about it?
Is there perhaps any workaround via TypeRig?

If not, is there at least a possibility to insert these "empty lines" automatically? (on selected nodes basis)?
![image](https://user-images.githubusercontent.com/68431285/134068326-b47128d5-1e0c-4989-90fb-d48d37d2af33.png)

Please let me know - thank you very much!
Vít

**Comment by kateliev** [comment]
Hi @Wirdek, regretfully there is no way to do that with **smart corners**. They are special kind of objects in FL API that behave like black boxes. You can either apply, remove them or set radius - nothing more. Even creating one using Python is not possible, you have to hijack the technology from a smart corner that the user has made manually somewhere in the font. If you are interested in working with smart corners on all masters and all the usual goodies you get with TypeRig (TR), please take a look at the **TR Filter Panel**. It allows you to set various radii across masters and apply them with a single click.

Now discussing the **TR Filter Panel** it can do some of the much needed operations on corners using other means/algorithms. It can round, mitre as well as ink-trap. Thus i encourage you to give it a try. Some tools might not work, as i have not revisited it for a long time, but if you report your problems i will fix the panel for you. Also if you are needing a way to "round-cap" stems as shown above, just let me know, i could have just the thing you need and could add it to the Filter panel if you are willing to wait a day or two. I am currently using that tech, but on a private project and can easily migrate the tools to the open-source.

**Comment by Wirdek** [comment]
Hi Vassil,
thank you very much for you quick reply and insights. I will definitely give the TR Filter Panel a try.

I am at the very early stage of designing a variable geometric-sans-superfamily. My idea is rougly this:
![image](https://user-images.githubusercontent.com/68431285/134145113-dd36d955-ad2c-422a-bcff-c7a61de720e5.png)


(+ I would like to also vary width and slant - not very far away from the Science Gothic of yours - Great font!)
Basically I'm looking for possibilities and ways how to achieve this and do it as modular and simple as possible.

But perhaps this is too "ambicious" as I dont have much experiences with variables (and also TypeRig) yet.
Therefore I may not use all the variable axes right now, but still - it would be great to have a good working "system" which alows me to expand it easily later on.

+ IMHO, the "smart corner" solution (if it worked as I suggested) would be very elegant and simple. 
One coud have several "identical" masters which would vary by the "corner-settings" only - it would be "live" and simple and very easily adjustable.

However it will get more complicated anyway as the strokes will not be perfectly mono-linear.
(at least heavy masters - the caps will be rather elliptical then circular).

Regarding "caps/terminals" I'm thinking of using elements - this might be the solution, what do you think?
![image](https://user-images.githubusercontent.com/68431285/134151397-6280bb5d-afb0-4a7f-94c7-69421220a0b3.png)

Anyway I'm looking forward to diving into the TypeRig soon - it surely has so many options and possibilities I have to learn :)

**Comment by Wirdek** [comment]
Hi Vassil,
I have just started a discussion on "smart cornes" and how to make them better - maybe you may be interested:

https://forum.fontlab.com/fontlab-7/smart-corners-better-smarter/

**Comment by Wirdek** [comment]
Hi Vassil,
thank you for your comments in FL formum. The TR Corner panel seems really quite powerful. 
However I'm having difficulties to run it.
At first I couldn't have found it under TypeRig panels whatsoever - so I've "refreshed scripts" in FL.

Now I'm not able to run TypeRig at all:
![image](https://user-images.githubusercontent.com/68431285/134649787-1df46389-cd00-49da-bcd9-1a758ef6cb3d.png)

What can I do? Please help.
Vít

**Comment by kateliev** [comment]
Hi Vit,

Please try the following - Do a fresh pull from the TR Git repo and drag the install.vfpy into your app (as if it was a font). This will open a revamped installed that i pushed a day ago and follow the instructions. If the Automatic install fails, use the manual options.

Aslo take a look here - the thread takes some wild directions, but explains a lot about installing TR: [https://forum.fontlab.com/fontlab-7/best-way-to-copy-a-specific-anchor/](https://forum.fontlab.com/fontlab-7/best-way-to-copy-a-specific-anchor/)

**Comment by kateliev** [comment]
Judging by the image above, to me it seems you are using the latest scripts but an older core?
